inspirehep/modules/literaturesuggest/dojson/fields/literature.py

```diff
@@ -91,12 +91,12 @@ def match_authors_initials(author_name):
 def categories(self, key, value):
     subject_list = [{
         "term": c,
-        "scheme": "arXiv"
+        "source": "user"
     } for c in value.split()]
-    if 'field_categories' in self:
-        self['field_categories'].extend(subject_list)
+    if 'inspire_categories' in self:
+        self['inspire_categories'].extend(subject_list)
     else:
-        self['field_categories'] = subject_list
+        self['inspire_categories'] = subject_list
     if 'arxiv_eprints' in self:
         self['arxiv_eprints'][0]['categories'] = value.split()
```

Comment:
These two lines are fundamentally wrong, because they expect `arxiv_eprints` to be there because the `arxiv_id` rule above already ran, but this is just not guaranteed to happen with DoJSON.